WeakAuras/Transmission.lua

```diff
@@ -977,43 +974,47 @@ function WeakAuras.MatchData(data, children)
   if not oldParent then return nil end
 
   info.oldData[0] = oldParent
+  info.newData[0] = data
   info.diffs[0] = WeakAuras.diff(oldParent, data)
   if info.diffs[0] then
     info.modified = info.modified + 1
   end
 
   if children then
     -- setup
-    info.deletedCount = #oldParent.controlledChildren
+    info.deleted = #oldParent.controlledChildren
     info.added = #children
-    local UIDtoID, IDtoData = {}, {}
+    local UIDtoID, IDtoData, IDtoIndex = {}, {}, {}
     for index, oldChildID in ipairs(oldParent.controlledChildren) do
       local oldChild = WeakAuras.GetData(oldChildID)
-      info.deleted[oldChildID] = true
       IDtoData[oldChildID] = oldChild
+      IDtoIndex[oldChildID] = index
       if oldChild.uid and not UIDtoID[oldChild.uid] then
         UIDtoID[oldChild.uid] = oldChildID
       end
     end
 
     -- confirm uid matches, find tentative matches
-    local tentativeMatches = {}
+    local tentativeMatches, confirmedNewMatches, confirmedOldMatches = {}, {}, {}
     for index, newChild in ipairs(children) do
       if newChild.uid then
         local matchedID = UIDtoID[newChild.uid]
         if matchedID then
```

Comment:
Even if newChild has uid, if there's no uid match, you probably want to try a id-match. That is the else branch of the newchild.uid branch should be run if there's no matchedId. (Though maybe check if the tentaitive old match has a uid and then reject that.)